contrib/gendocs.py

```diff
@@ -0,0 +1,297 @@
+def cleanup():
+    try:
+        shutil.move("test/tests.asciidoc", "test/README.asciidoc")
+    except BaseException:
+        pass
+    try:
+        shutil.move("index.asciidoc", "README.asciidoc")
+    except BaseException:
+        pass
+    try:
+        shutil.move("VIMTOKAK.asciidoc", "VIMTOKAK")
+    except BaseException:
+        pass
+
+
+atexit.register(cleanup)
+
+
+# Antora fails if the repo contains broken symbolic links.
+shutil.rmtree("libexec", ignore_errors=True)
+
+# Canonical Antora paths.
+# See: https://docs.antora.org/antora/latest/standard-directories.
+#      https://docs.antora.org/antora/latest/root-module-directory.
+os.makedirs("modules/ROOT/images", exist_ok=True)
+os.makedirs("modules/ROOT/pages", exist_ok=True)
+
+# The file name is used for navigation links.
+# Update so it reflects the content.
+shutil.move("test/README.asciidoc", "test/tests.asciidoc")
+
+# Update the filename so it reflects the content.
+# The filename is used for navigation links.
+shutil.move("README.asciidoc", "index.asciidoc")
+
+# A useful documentation page.
+# Add the `.asciidoc` extension to include it into the result.
+shutil.move("VIMTOKAK", "VIMTOKAK.asciidoc")
+
+# Put necessary images to the Antora canonical directory.
+# See: https://docs.antora.org/antora/latest/images-directory.
+for gif_file in glob.glob("doc/*.gif"):
+    shutil.copy(gif_file, "modules/ROOT/images/")
+
+
+# Fix links according to the Antora specification.
+# See: https://docs.antora.org/antora/latest/page/xref.
+def fix_links(path):
+    @dataclass
+    class Link:
+        path: Optional[str]
+        file: Optional[str]
+        fragment: Optional[str]
+        title: str
+
+        def __init__(self, path, file, fragment, title):
+            self.path = path
+            self.file = file
+            self.fragment = fragment
+            self.title = title
+
+    def dropwhile(predicate, string):
+        return "".join(itertools.dropwhile(predicate, string))
+
+    def dropwhileright(predicate, string):
+        return "".join(reversed(list(itertools.dropwhile(predicate, reversed(string)))))
+
+    def takewhile(predicate, string):
+        return "".join(itertools.takewhile(predicate, string))
+
+    def untag(string):
+        no_opening = dropwhile(lambda c: c == "<", string)
+        no_closing = dropwhileright(lambda c: c == ">", no_opening)
+        return no_closing
+
+    def parse(string):
+        untagged = untag(string)
+        prefix, title = untagged.split(",", 1)
+        title = title.strip()
+        fragment = dropwhile(lambda c: c != "#", prefix)
+        fragment = fragment if fragment else None
+        fragmentless = takewhile(lambda c: c != "#", prefix)
+        segments = fragmentless.split("/")
+        path, file = (
+            ("/".join(segments[:-1]), segments[-1])
+            if "/" in fragmentless
+            else (None, fragmentless)
+        )
+        return Link(path, file, fragment, title)
```

Comment:
Actually GitHub is decent at rendering asciidoc, they also support the cross references.
Though the dedicated site definitely looks nicer and is more discoverable.
Maybe we can add a link to it on the website.
It does have a large overlap with github so I'm not sure it will gain a lot of momentum.
But it's nice to have a replacement we can custommize as we add more docs